### PR TITLE
Rename lake::GroupAssigner to lake::LocationProvider

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

### CODE REVIEW

**Review by imay — DISMISSED** [review]

File: be/src/storage/CMakeLists.txt [hunk 1]
```diff
@@ -162,7 +162,7 @@ add_library(Storage STATIC
     lake/delta_writer.cpp
     lake/general_tablet_writer.cpp
     lake/rowset.cpp
-    lake/starlet_group_assigner.cpp
+        lake/starlet_location_provider.cpp
```
Comment (wyb) [inline be/src/storage/CMakeLists.txt:165]:
```suggestion
    lake/starlet_location_provider.cpp
```

File: be/src/runtime/exec_env.cpp [hunk 2]
```diff
@@ -78,23 +78,23 @@
...
-    explicit FixedGroupAssigner(std::string path) : _path(std::move(path)) {
+    explicit FixedLocationProvider(std::string path) : _path(std::move(path)) {
         if (_path.back() == '/') _path.pop_back();
     }
 
-    ~FixedGroupAssigner() override = default;
+    ~FixedLocationProvider() override = default;
 
-    std::string get_fs_prefix() override { return "posix://"; }
+    std::string fs_scheme() const override { return "posix://"; }
```
Comment (wyb) [inline be/src/runtime/exec_env.cpp:89]: schema does not include `://`.

File: be/src/runtime/exec_env.cpp [hunk 3]
```diff
@@ -273,12 +273,13 @@ Status ExecEnv::_init(const std::vector<StorePath>& store_paths) {
             exit(-1);
         }
 #ifndef USE_STAROS
-        _lake_group_assigner = new FixedGroupAssigner(_store_paths.front().path);
+        _lake_location_provider = new FixedLocationProvider(_store_paths.front().path);
 #else
         _lake_group_assigner = new lake::StarletGroupAssigner();
```
Comment (wyb) [inline be/src/runtime/exec_env.cpp:278]:
```suggestion
        _lake_location_provider = new lake:: StarletLocationProvider();
```

File: be/src/storage/lake/starlet_location_provider.cpp [hunk 4]
```diff
@@ -18,19 +18,19 @@ static std::string normalize_group(const std::string& group) {
     return group.substr(0, group.length() - 1);
 }
 
-std::string StarletGroupAssigner::get_fs_prefix() {
+std::string StarletGroupAssigner::fs_scheme() const {
```
Comment (wyb) [inline be/src/storage/lake/starlet_location_provider.cpp:21]:
```suggestion
std::string StarletLocationProvider::fs_scheme() const {
```

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/storage/CMakeLists.txt (hunk 1 above)
Comment (sduzh) [reply be/src/storage/CMakeLists.txt:165]: updated

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/runtime/exec_env.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/runtime/exec_env.cpp]: I've removed this method.

File: be/src/runtime/exec_env.cpp (hunk 3 above)
Comment (sduzh) [reply be/src/runtime/exec_env.cpp]: fixed

File: be/src/storage/lake/starlet_location_provider.cpp (hunk 4 above)
Comment (sduzh) [reply be/src/storage/lake/starlet_location_provider.cpp]: fixed

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

File: be/src/storage/lake/tablet_manager.cpp [hunk 5]
```diff
@@ -137,26 +142,25 @@ Status TabletManager::create_tablet(const TCreateTabletReq& req) {
...
 
 Status TabletManager::drop_tablet(int64_t tablet_id) {
     std::vector<std::string> objects;
     const auto tablet_metadata_prefix = fmt::format("tbl_{:016X}_", tablet_id);
     const auto txnlog_prefix = fmt::format("txn_{:016X}_", tablet_id);
 
-    // get group path
-    ASSIGN_OR_RETURN(auto group_path, _group_assigner->get_group(tablet_id));
-    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(_group_assigner->get_fs_prefix()));
+    ASSIGN_OR_RETURN(auto root_path, _location_provider->root_location(tablet_id));
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.cpp:162]: Need `staros://` prefix.

File: be/src/storage/lake/tablet_manager.cpp [hunk 6]
```diff
@@ -271,23 +272,22 @@ StatusOr<TabletMetadataIter> TabletManager::list_tablet_metadata(int64_t tablet_
         prefix = "tbl_";
     }
 
-    ASSIGN_OR_RETURN(auto group, _group_assigner->get_group(tablet_id));
-    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(_group_assigner->get_fs_prefix()));
+    ASSIGN_OR_RETURN(auto root, _location_provider->root_location(tablet_id));
+    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(root));
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.cpp:276]: Need `staros://` prefix.

File: be/src/storage/lake/tablet_manager.cpp [hunk 7]
```diff
@@ -378,18 +377,17 @@ StatusOr<TxnLogIter> TabletManager::list_txn_log(int64_t tablet_id, bool filter_
         prefix = "txn_";
     }
 
-    ASSIGN_OR_RETURN(auto group, _group_assigner->get_group(tablet_id));
-    // get group path
-    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(_group_assigner->get_fs_prefix()));
+    ASSIGN_OR_RETURN(auto root, _location_provider->root_location(tablet_id));
+    ASSIGN_OR_RETURN(auto fs, FileSystem::CreateSharedFromString(root));
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.cpp:381]: Need `staros://` prefix.

File: be/src/storage/lake/tablet_manager.cpp (hunk 5 above)
Comment (sduzh) [reply be/src/storage/lake/tablet_manager.cpp:162]: Already included in `root_location()`

Comment (sduzh) [reply be/src/storage/lake/tablet_manager.cpp:162]: I'll fix this in next pr.

**Review by wyb — APPROVED** [review]

File: be/src/storage/lake/tablet_manager.cpp (hunk 6 above)
Comment (sduzh) [reply be/src/storage/lake/tablet_manager.cpp:276]: ditto

**Pull request merged**